ocaml/xapi/rrdd_proxy.ml

```diff
@@ -87,33 +86,38 @@ let get_vm_rrd_forwarder (req : Http.Request.t) (s : Unix.file_descr) _ =
         let is_unarchive_request =
           List.mem_assoc Constants.rrd_unarchive query
         in
-        let is_owner_online owner = Db.is_valid_ref __context owner in
-        let is_xapi_initialising = List.mem_assoc "dbsync" query in
+        let metrics_at () =
+          let ( let* ) = Option.bind in
+          let owner_of vm =
+            let owner = Db.VM.get_resident_on ~__context ~self:vm in
+            let is_xapi_initialising = List.mem_assoc "dbsync" query in
+            let is_available = not is_xapi_initialising in
```

Comment:
I think this was like this before, but `is_xapi_initializing` refers to the local XAPI, not the remote one, right?

Reply:
This endpoint might be called from a host in the pool while xapi is being initialized. In that case the dbsync parameter is present in the query